cmd/detectExecuteScan.go

```diff
@@ -605,17 +606,25 @@ func addDetectArgsImages(args []string, config detectExecuteScanOptions, utils d
 	args = append(args, "--detect.docker.passthrough.imageinspector.service.start=false")
 	args = append(args, "--detect.docker.passthrough.output.include.squashedimage=false")
 
+	port := ""
 	switch config.ScanContainerDistro {
 	case "ubuntu":
-		args = append(args, "--detect.docker.passthrough.imageinspector.service.url=http://localhost:8082")
+		port = "8082"
 	case "centos":
-		args = append(args, "--detect.docker.passthrough.imageinspector.service.url=http://localhost:8081")
+		port = "8081"
 	case "alpine":
-		args = append(args, "--detect.docker.passthrough.imageinspector.service.url=http://localhost:8080")
+		port = "8080"
 	default:
 		return nil, fmt.Errorf("unknown container distro %q", config.ScanContainerDistro)
 	}
 
+	host := fmt.Sprintf("inspector-%s", config.ScanContainerDistro)
+	// if we run in a Kubernetes environment, we can reach the sidecar using "localhost"
+	if os.Getenv("ON_K8S") == "true" {
```

Comment:
Shouldn't we get rid of such ifs instead of adding them more and more. Doesn't that defeat the purpose of the jenkins-library if the steps will have to know where they run?